src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -3360,19 +3360,18 @@ internal enum AddressKind
             }
 
             // while readonly fields have home it is not valid to refer to it when not constructing.
-            if (!TypeSymbol.Equals(field.ContainingType, method.ContainingType, TypeCompareKind.ConsiderEverything2))
+            if (!TypeSymbol.Equals(field.ContainingType, method.ContainingType, TypeCompareKind.ConsiderEverything))
```

Comment:
>ConsiderEverything [](start = 96, length = 18)

This option doesn't look right, why would difference in nullable annotations, or dynamic, or native int, or tuple names should make a difference? I think we should ignore all options that could be ignored.

Reply:
I think the problem can be demonstrated with this failing test:
```
        [Fact]
        public void Test()
        {
            string source = @"
public class C
{
    public static void Main()
    {
        System.Console.WriteLine(C1<int>.F1.content);
        System.Console.WriteLine(C2<int>.F1.content);
    }
}

public struct Container
{
    public int content;
}

class C1<T>
{
    public static readonly Container F1;
    
    static C1()
    {
        C1<T>.F1.content = 2;
    }
}


#nullable enable

class C2<T>
{
    public static readonly Container F1;
    
    static C2()
    {
        C2<T>.F1.content = 2;
    }
}
";
            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview, options: TestOptions.DebugExe);
            comp.VerifyEmitDiagnostics();
            CompileAndVerify(comp, expectedOutput: 
@"
2
2
");
        }
```

---
In reply to: [460448404](https://github.com/dotnet/roslyn/pull/46323#discussion_r460448404) [](ancestors = 460448404)

Reply:
Thanks much for the test scenario. Added and fixed.